### PR TITLE
Copy widget to dashboard: filter dashboards by permissions

### LINKED ISSUE #7331: Filter list of dashboards to copy widgets to
## Expected Behavior
When clicking on copy to dashboard the user should get a list of dashboards where he has
edit permissions to.

## Current Behavior
The user gets a list of dashboards where he has read permissions to.

## Steps to Reproduce (for bugs)
1. Create Role with read dashboards and edit dashoards.
2. Assign Role to User.
3. Log in as user.
4. Go to search and copy a widget to dashboard and look at the list.

## Your Environment
* Graylog Version: 3.2 RC1

### PR DESCRIPTION
**Pull request by janheise**
## Description

Prior to this change, the list of dashboards in the copy-to dialog for a widget contained all dashboards. Now only those dashboards are listed that the user actually has permissions to change.

fixes #7331 

## How Has This Been Tested?

manually tested by sharing dashboard between users and logging in/opening the copy dialog

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by linuspahl — COMMENTED** [review]
I tested this PR and it worked as expected.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/DashboardsResource.java [hunk 1]
```diff
@@ -95,6 +96,34 @@ public PageListResponse<ViewSummaryDTO> views(@ApiParam(name = "page") @QueryPar
                                                   @ApiParam(name = "query") @QueryParam("query") String query,
                                                   @ApiParam(name = "filters") @QueryParam("filters") List<String> filters,
                                                   @Context SearchUser searchUser) {
+        return views(page, perPage, sortField, order, query, searchUser::canReadView, filters, searchUser);
+    }
+
+    @GET
+    @ApiOperation("Get a list of all dashboards a user can change")
+    @Path("/canUpdate")
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/DashboardsResource.java:104]: I would suggest we merge this into the `views` method and just provide a query parameter for the options. IMO it is some kind of a filter for that endpoint.

**Review by dennisoelkers — DISMISSED** [review]

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/DashboardsResource.java (hunk 1 above)
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/DashboardsResource.java:104]: will do

**Review by linuspahl — APPROVED** [review]
Still works as expected.

**Pull request merged**